### PR TITLE
Feature/add claim status

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java [hunk 1]
```diff
@@ -18,31 +22,42 @@
     private final EligibilityClient client;
     private final EligibilityStatusCalculator eligibilityStatusCalculator;
 
+    private final Map<EligibilityStatus, ClaimStatus> eligibilityStatusMapper = ImmutableMap.of(
+            EligibilityStatus.ELIGIBLE, ClaimStatus.NEW,
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java]: We're missing EligibilityStatus.INELIGIBLE -> REJECTED

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java [hunk 2]
```diff
@@ -18,31 +22,42 @@
     private final EligibilityClient client;
     private final EligibilityStatusCalculator eligibilityStatusCalculator;
 
+    private final Map<EligibilityStatus, ClaimStatus> eligibilityStatusMapper = ImmutableMap.of(
```
Comment (YetAnotherMatt) [inline api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java]: This might be better called 'statusMap'?

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java [hunk 3]
```diff
@@ -36,19 +38,25 @@
     @Mock
     private EligibilityStatusCalculator eligibilityStatusCalculator;
 
-    @Test
-    public void shouldSaveNewClaimant() {
+    @ParameterizedTest(name = "Should save claimant with claim status set to {1} when eligibility status is {0}")
+    @CsvSource({
+            "ELIGIBLE, NEW",
```
Comment (YetAnotherMatt) [inline api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java]: we need INELIGIBLE here too. Can we add a test that uses an EligibilityStatus EnumSource and confirms that the persisted claimStatus is not null?

File: db/src/main/resources/db/migration/V1_006__add_claim_status_column.sql [hunk 4]
```diff
@@ -0,0 +1,7 @@
+alter table claimant add column claim_status varchar(20);
+
+update claimant set claim_status = 'ACTIVE' where claim_status is null and eligibility_status = 'ELIGIBLE';
+
+update claimant set claim_status = 'REJECTED' where claim_status is null and eligibility_status != 'ELIGIBLE';
```
Comment (YetAnotherMatt) [inline db/src/main/resources/db/migration/V1_006__add_claim_status_column.sql]: this could actually just be `update claimant set claim_status = 'REJECTED' where claim_status is null;`

File: db/src/main/resources/db/migration/V1_006__add_claim_status_column.sql [hunk 5]
```diff
@@ -0,0 +1,7 @@
...
+
+alter table claimant alter column claim_status set not null;
```
Comment (YetAnotherMatt) [inline db/src/main/resources/db/migration/V1_006__add_claim_status_column.sql]: I think we'll need an index on claim_status

Comment (dwybourn) [reply db/src/main/resources/db/migration/V1_006__add_claim_status_column.sql]: Added index

File: db/src/main/resources/db/migration/V1_006__add_claim_status_column.sql (hunk 4 above)
Comment (dwybourn) [reply db/src/main/resources/db/migration/V1_006__add_claim_status_column.sql]: Updated.

File: api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java (hunk 3 above)
Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/claimant/service/ClaimServiceTest.java]: Added missing case.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java (hunk 1 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java]: Added missing case.

File: api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java (hunk 2 above)
Comment (dwybourn) [reply api/src/main/java/uk/gov/dhsc/htbhf/claimant/service/ClaimService.java]: Renamed.
